--- qt-copy/tools/assistant/main.cpp.niedakh	2005-05-13 16:39:06.000000000 +0200
+++ qt-copy/tools/assistant/main.cpp	2005-05-14 14:34:40.000000000 +0200
@@ -272,11 +272,12 @@
         resourceDir = QLibraryInfo::location(QLibraryInfo::TranslationsPath);
 
     QTranslator translator( 0 );
-    translator.load( QLatin1String("assistant_") + QLocale::system().name().toLower(), resourceDir );
+    QLocale locale = QLocale::system();
+    translator.load( QLatin1String("assistant"), resourceDir + QLatin1String("/") + QLocale::languageToString(locale.language())  + QLatin1String("/LC_MESSAGES") );
     a.installTranslator( &translator );
 
     QTranslator qtTranslator( 0 );
-    qtTranslator.load( QLatin1String("qt_") + QLocale::system().name().toLower(), resourceDir );
+    qtTranslator.load( QLatin1String("qt"), resourceDir + QLatin1String("/") + QLocale::languageToString(locale.language()) + QLatin1String("/LC_MESSAGES") );
     a.installTranslator( &qtTranslator );
 
     Config *conf = Config::loadConfig( profileName );
