--- qt-x11-opensource-src-4.5.0/qmake/Makefile.unix~	2009-02-25 22:09:16.000000000 +0100
+++ qt-x11-opensource-src-4.5.0/qmake/Makefile.unix	2009-03-07 21:16:08.902865896 +0100
@@ -161,7 +161,7 @@
 	$(CXX) -c -o $@ $(CXXFLAGS) $(SOURCE_PATH)/src/corelib/io/qurl.cpp
 
 qstring.o: $(SOURCE_PATH)/src/corelib/tools/qstring.cpp
-	$(CXX) -c -o $@ $(CXXFLAGS) $(SOURCE_PATH)/src/corelib/tools/qstring.cpp
+	$(CXX) -c -o $@ $(CXXFLAGS) -O1 $(SOURCE_PATH)/src/corelib/tools/qstring.cpp
 
 qlocale.o: $(SOURCE_PATH)/src/corelib/tools/qlocale.cpp
 	$(CXX) -c -o $@ $(CXXFLAGS) $(SOURCE_PATH)/src/corelib/tools/qlocale.cpp
