--- qt-x11-opensource-src-4.1.0/src/qbase.pri.orig	2005-12-16 12:51:29.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/src/qbase.pri	2006-02-12 22:11:51.000000000 +0100
@@ -32,7 +32,6 @@
 CONFIG		+= qt warn_on depend_includepath
 CONFIG          += qmake_cache target_qt 
 CONFIG          -= fix_output_dirs
-!macx-xcode:CONFIG += debug_and_release
 
 contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
 contains(QT_CONFIG, largefile):CONFIG += largefile
--- qt-x11-opensource-src-4.1.0/src/plugins/qpluginbase.pri.orig	2005-12-16 12:51:35.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/src/plugins/qpluginbase.pri	2006-02-12 22:14:12.000000000 +0100
@@ -1,7 +1,6 @@
 TEMPLATE = lib
 CONFIG += qt plugin
 
-!win32-msvc:!macx-xcode:CONFIG += debug_and_release
 !debug_and_release|build_pass {
    CONFIG(debug, debug|release): {
       unix:TARGET = $$member(TARGET, 0)_debug
--- qt-x11-opensource-src-4.1.0/tools/assistant/lib/lib.pro.orig	2006-02-12 22:08:26.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/tools/assistant/lib/lib.pro	2006-02-12 22:09:03.000000000 +0100
@@ -4,7 +4,7 @@
 VERSION                = 1.0
 
 CONFIG                += qt warn_on
-CONFIG                += dll debug_and_release
+CONFIG                += dll
 mac:unix:CONFIG       += explicitlib
 
 HEADERS         = qassistantclient.h
--- qt-x11-opensource-src-4.1.0/tools/designer/src/lib/lib.pro.orig	2005-12-16 12:52:20.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/tools/designer/src/lib/lib.pro	2006-02-12 23:01:41.000000000 +0100
@@ -2,7 +2,7 @@
 TARGET=QtDesigner
 QT += xml
 contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
-CONFIG += qt debug_and_release
+CONFIG += qt
 DESTDIR = ../../../../lib
 DLLDESTDIR = ../../../../bin
 
--- qt-x11-opensource-src-4.1.0/tools/designer/src/components/lib/lib.pro.orig	2005-12-16 12:52:25.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/tools/designer/src/components/lib/lib.pro	2006-02-12 23:03:35.000000000 +0100
@@ -1,7 +1,7 @@
 TEMPLATE = lib
 TARGET = QtDesignerComponents
 contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
-CONFIG += qt debug_and_release depend_prl
+CONFIG += qt depend_prl
 DESTDIR = $$QT_BUILD_TREE/lib
 DLLDESTDIR = $$QT_BUILD_TREE/bin
 
--- qt-x11-opensource-src-4.1.0/tools/designer/src/components/component.pri.orig	2005-12-16 12:52:20.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/tools/designer/src/components/component.pri	2006-02-12 22:16:03.000000000 +0100
@@ -1,5 +1,4 @@
 
-CONFIG += debug_and_release
 contains(TEMPLATE,lib) {
    CONFIG(debug, debug|release) {
       unix:TARGET = $$member(TARGET, 0)_debug
--- qt-x11-opensource-src-4.1.0/tools/designer/src/plugins/plugins.pri.orig	2005-12-16 12:52:20.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/tools/designer/src/plugins/plugins.pri	2006-02-12 23:05:32.000000000 +0100
@@ -1,4 +1,4 @@
-CONFIG += designer debug_and_release
+CONFIG += designer
 DESTDIR = $$QT_BUILD_TREE/plugins/designer
 contains(TEMPLATE,lib) {
    CONFIG(debug, debug|release) {
--- qt-x11-opensource-src-4.1.0/demos/arthurplugin/arthurplugin.pro.orig	2005-12-16 12:51:46.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/demos/arthurplugin/arthurplugin.pro	2006-02-13 00:14:21.000000000 +0100
@@ -1,7 +1,7 @@
 
 QTDIR       = $$QT_SOURCE_TREE
 
-CONFIG      += designer plugin debug_and_release
+CONFIG      += designer plugin
 TEMPLATE    = lib
 DESTDIR     = $$QT_BUILD_TREE/plugins/designer
 
--- qt-x11-opensource-src-4.1.1/examples/designer/containerextension/containerextension.pro.orig	2006-02-16 11:51:28.000000000 +0100
+++ qt-x11-opensource-src-4.1.1/examples/designer/containerextension/containerextension.pro	2006-02-23 12:01:07.461084000 +0100
@@ -1,5 +1,5 @@
 TEMPLATE = lib
-CONFIG  += designer plugin debug_and_release
+CONFIG  += designer plugin
 DESTDIR  = $$QT_BUILD_TREE/plugins/designer
 
 CONFIG(debug, debug|release) {
--- qt-x11-opensource-src-4.1.0/examples/designer/customwidgetplugin/customwidgetplugin.pro.orig	2005-12-16 12:52:13.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/examples/designer/customwidgetplugin/customwidgetplugin.pro	2006-02-13 00:16:43.000000000 +0100
@@ -1,4 +1,4 @@
-CONFIG      += designer plugin debug_and_release
+CONFIG      += designer plugin
 TEMPLATE    = lib
 DESTDIR     = $$QT_BUILD_TREE/plugins/designer
 
--- qt-x11-opensource-src-4.1.1/examples/designer/taskmenuextension/taskmenuextension.pro.orig	2006-02-16 11:51:28.000000000 +0100
+++ qt-x11-opensource-src-4.1.1/examples/designer/taskmenuextension/taskmenuextension.pro	2006-02-23 12:07:32.997178500 +0100
@@ -1,5 +1,5 @@
 TEMPLATE = lib
-CONFIG  += designer plugin debug_and_release
+CONFIG  += designer plugin
 DESTDIR  = $$QT_BUILD_TREE/plugins/designer
 
 CONFIG(debug, debug|release) {
--- qt-x11-opensource-src-4.1.0/examples/designer/worldtimeclockplugin/worldtimeclockplugin.pro.orig	2005-12-16 12:52:13.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/examples/designer/worldtimeclockplugin/worldtimeclockplugin.pro	2006-02-13 00:18:50.000000000 +0100
@@ -1,4 +1,4 @@
-CONFIG      += designer plugin debug_and_release
+CONFIG      += designer plugin
 TEMPLATE    = lib
 DESTDIR     = $$QT_BUILD_TREE/plugins/designer
 
--- qt-x11-opensource-src-4.1.1/tools/designer/src/uitools/uitools.pro.orig	2006-02-16 11:51:38.000000000 +0100
+++ qt-x11-opensource-src-4.1.1/tools/designer/src/uitools/uitools.pro	2006-02-23 12:11:12.242880500 +0100
@@ -4,7 +4,6 @@
 CONFIG += qt staticlib
 DESTDIR = ../../../../lib
 DLLDESTDIR = ../../../../bin
-CONFIG += debug_and_release
 CONFIG(debug, debug|release) {
     unix:TARGET = $$member(TARGET, 0)_debug
     else:TARGET = $$member(TARGET, 0)d
