--- qt-x11-opensource-src-4.3.1/src/sql/kernel/qsqldatabase.cpp.orig	2007-08-07 15:08:33.000000000 +0200
+++ qt-x11-opensource-src-4.3.1/src/sql/kernel/qsqldatabase.cpp	2007-08-29 17:28:40.473284033 +0200
@@ -69,7 +69,10 @@
 #include "../drivers/sqlite2/qsql_sqlite2.h"
 #endif
 #ifdef QT_SQL_IBASE
+/* workaround for ODBC/Firebird types conflict */
+#define SCHAR SCHAR_IBASE
 #include "../drivers/ibase/qsql_ibase.h"
+#undef SCHAR
 #endif
 
 #include "qdebug.h"
