--- qt-everywhere-opensource-src-4.8.0/tools/qvfb/qvfb.pro~	2011-12-08 06:06:02.000000000 +0100
+++ qt-everywhere-opensource-src-4.8.0/tools/qvfb/qvfb.pro	2011-12-15 20:13:00.159241557 +0100
@@ -38,8 +38,6 @@
 
 !embedded {
     DEFINES         += QT_NO_QWS_SIGNALHANDLER
-    HEADERS         += ../../src/gui/embedded/qlock_p.h
-    SOURCES         += ../../src/gui/embedded/qlock.cpp
 }
 
 include(../shared/deviceskin/deviceskin.pri)
