--- qt-x11-opensource-src-4.3.3/src/sql/kernel/qsqldatabase.cpp.orig	2007-12-04 17:42:49.000000000 +0100
+++ qt-x11-opensource-src-4.3.3/src/sql/kernel/qsqldatabase.cpp	2008-01-04 18:21:02.075809521 +0100
@@ -74,8 +74,12 @@
 #undef SQL_TYPE_DATE
 #undef SQL_DATE
 #define SCHAR IBASE_SCHAR  // avoid clash with ODBC (older versions of ibase.h with Firebird)
+#define SLONG IBASE_SLONG
+#define ULONG IBASE_ULONG
 #include "../drivers/ibase/qsql_ibase.h"
 #undef SCHAR
+#undef SLONG
+#undef ULONG
 #endif
 
 #include "qdebug.h"
