--- qt-everywhere-opensource-src-4.7.0/tools/tools.pro.orig	2010-09-21 21:30:01.308429119 +0200
+++ qt-everywhere-opensource-src-4.7.0/tools/tools.pro	2010-09-21 21:30:39.836139945 +0200
@@ -13,6 +13,7 @@
                       pixeltool \
                       porting \
                       qtestlib \
+		      qvfb \
                       qttracereplay
             contains(QT_EDITION, Console) {
                 SUBDIRS += designer/src/uitools     # Linguist depends on this
