--- qt-x11-opensource-src-4.1.0/src/corelib/arch/alpha/arch/qatomic.h.orig	2005-12-16 12:52:24.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/src/corelib/arch/alpha/arch/qatomic.h	2006-02-18 15:32:28.721401500 +0100
@@ -87,7 +87,7 @@
     return ret;
 }
 
-inline int q_atomic_test_and_set_release_ptr(volatile void *ptr, void *expected, void *newval)
+inline int q_atomic_test_and_set_ptr(volatile void *ptr, void *expected, void *newval)
 {
     register void *ret;
     asm volatile("1:\n"
