--- src/gui/kernel/qapplication_x11.cpp	2007-12-18 11:52:08.000000000 +0100
+++ src/gui/kernel/qapplication_x11.cpp-new	2008-02-23 16:29:23.550847092 +0100
@@ -1699,7 +1699,7 @@
                 (PtrXcursorLibraryLoadCursor) xcursorLib.resolve("XcursorLibraryLoadCursor");
         }
 #else
-        ptrXcursorLibraryLoadCursor = XcursorLibraryLoadCursor;
+        X11->ptrXcursorLibraryLoadCursor = (PtrXcursorLibraryLoadCursor) xcursorLib.resolve("XcursorLibraryLoadCursor");
 #endif // QT_RUNTIME_XCURSOR
 #endif // QT_NO_XCURSOR
 
